dask_sql/context.py

```diff
@@ -845,15 +845,19 @@ def _get_ral(self, sql):
     def _compute_table_from_rel(self, rel: "LogicalPlan", return_futures: bool = True):
         dc = RelConverter.convert(rel, context=self)
 
-        # Optimization might remove some alias projects. Make sure to keep them here.
-        select_names = [field for field in rel.getRowType().getFieldList()]
-
         if rel.get_current_node_type() == "Explain":
             return dc
         if dc is None:
             return
 
+        # Optimization might remove some alias projects. Make sure to keep them here.
+        select_names = [field for field in rel.getRowType().getFieldList()]
+
         if select_names:
+            cc = dc.column_container
+
+            select_names = select_names[: len(cc.columns)]
```

Comment:
Do you know why `getFieldList` and `getFieldNames` are returning an extra field that we need to filter out here (and in `fix_column_to_row_type`)? I want to believe there's something we could do on the Rust end to avoid needing these workarounds in the Python code, but okay with filing that as a follow up issue to address.

cc @jdye64 if you have any idea what could be going on here

Reply:
Yeah it seems like it's treating left semi and left anti joins like left joins where the column merged on from both left and right tables is included in the result whereas in leftsemi and leftanti joins only the left hand column should be returned

Reply:
Would you mind filing an issue around this and linking it here so we could follow up on this later if possible?

Reply:
#1192